tests/e2e/vmgroupmisc_test.go

```diff
@@ -0,0 +1,269 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test suite contains miscellaneous tests to verify behavior of non-default vmgroup
+
+// +build runonce
+
+package e2e
+
+import (
+	"log"
+
+	admincliconst "github.com/vmware/docker-volume-vsphere/tests/constants/admincli"
+	"github.com/vmware/docker-volume-vsphere/tests/constants/properties"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/admincli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/esx"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	vmGroupName = "vg_basictest"
+)
+
+type vgBasicSuite struct {
+	config      *inputparams.TestConfig
+	volumeNames []string
+}
+
+func (s *vgBasicSuite) SetUpSuite(c *C) {
+	basicVG := "basicVGTest"
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping vmgroupbasic tests.")
+	}
+	admincli.ConfigInit(s.config.EsxHost)
+
+	// Verify DB successfully initialized
+	c.Assert(admincli.GetDBmode(s.config.EsxHost), Equals, admincliconst.DBSingleNode, Commentf("Failed to init the DB mode on ESX -  .", s.config.EsxHost))
+	s.volumeNames = []string{inputparams.GetUniqueVolumeName(basicVG), inputparams.GetUniqueVolumeName(basicVG)}
+}
+
+func (s *vgBasicSuite) SetUpTest(c *C) {
+	// Creating non-default vmgroup only if it does not exists
+	if admincli.IsVmgroupPresent(s.config.EsxHost, vmGroupName) {
+		return
+	}
+	admincli.CreateVMgroup(s.config.EsxHost, vmGroupName, s.config.DockerHostNames[0], s.config.Datastores[0])
+
+	// Verify if vmgroup exists
+	isVmgroupAvailable := admincli.IsVmgroupPresent(s.config.EsxHost, vmGroupName)
+	c.Assert(isVmgroupAvailable, Equals, true, Commentf("vmgroup ls command does not lists the vmgroup %s .", vmGroupName))
+
+	// Verify vm belongs to vmgroup
+	isVMPartofVg := admincli.IsVMInVmgroup(s.config.EsxHost, s.config.DockerHostNames[0], vmGroupName)
+	c.Assert(isVMPartofVg, Equals, true, Commentf("VM %s does not belong to vmgroup %s .", s.config.DockerHostNames[0], vmGroupName))
+}
+
+func (s *vgBasicSuite) TearDownSuite(c *C) {
+	admincli.RemoveVMFromVMgroup(s.config.EsxHost, vmGroupName, s.config.DockerHostNames[0])
+
+	// Verify vm does not belongs to vmgroup
+	isVMPartofVg := admincli.IsVMInVmgroup(s.config.EsxHost, s.config.DockerHostNames[0], vmGroupName)
```

Comment:
I am removing the vm from vmgroup so I am just making sure removal was successful.